restapi/src/test/java/io/stargate/web/docsapi/service/DocumentServiceTest.java

```diff
@@ -527,6 +528,108 @@ void testGetDocPathFields() throws JsonProcessingException {
         .isEqualTo("[{\"a\":{\"d\":{\"c\":3,\"z\":true}}},{\"a\":{\"f\":{\"c\":100}}}]");
   }
 
+  @Test
+  void testGetDocPathPlainFields() throws JsonProcessingException {
+    final String id = "id2";
+    withQuery(table, selectAll("WHERE key = ? AND p0 = ? AND p1 = ?"), id, "a", "d")
+        .returning(
+            ImmutableList.of(
+                row(id, 3.0, "a", "d", "c"),
+                row(id, true, "a", "d", "z"),
+                row(id, "zz", "a", "d", "y")));
+
+    DocumentResponseWrapper<Map<String, ?>> r =
+        getDocPath(id, null, "[\"c\", \"z\"]", ImmutableList.of(p("a"), p("d")));
+    assertThat(r.getDocumentId()).isEqualTo(id);
+    assertThat(r.getPageState()).isNull();
+    assertThat(r.getData()).isEqualTo(m("c", 3, "z", true));
+
+    // selecting missing field results in no docs found
+    assertThat(
+            resource
+                .getDocPath(
+                    headers,
+                    uriInfo,
+                    authToken,
+                    keyspace.name(),
+                    table.name(),
+                    id,
+                    ImmutableList.of(p("a"), p("d")),
+                    null,
+                    "[\"x\"]",
+                    null,
+                    null,
+                    false,
+                    false,
+                    request)
+                .getStatus())
+        .isEqualTo(Status.NOT_FOUND.getStatusCode());
+
+    withQuery(table, selectAll("WHERE key = ?"), id)
+        .returning(ImmutableList.of(row(id, 3.0, "x"), row(id, true, "y"), row(id, "zz", "z")));
+
+    r = getDocPath(id, null, "[\"x\", \"z\"]", ImmutableList.of());
+    assertThat(r.getDocumentId()).isEqualTo(id);
+    assertThat(r.getPageState()).isNull();
+    assertThat(r.getData()).isEqualTo(m("x", 3, "z", "zz"));
+  }
+
+  @Test
+  void testGetDocPathPlainFieldsPaged() throws JsonProcessingException {
+    final String id = "id2";
+    withQuery(table, selectAll("WHERE key = ? AND p0 = ? AND p1 = ? ALLOW FILTERING"), id, "a", "b")
+        .returning(
+            ImmutableList.of(
+                row(id, 3.0, "a", "b", "c1"),
+                row(id, true, "a", "b", "c2"),
+                row(id, "zz", "a", "b", "d")));
+
+    DocumentResponseWrapper<List<Map<String, ?>>> r =
+        getDocPath(id, null, "[\"c1\", \"d\"]", ImmutableList.of(p("a"), p("b")), 1, null, false);
+    assertThat(r.getDocumentId()).isEqualTo(id);
+    assertThat(r.getPageState()).isNull();
+    assertThat(r.getData()).isEqualTo(ImmutableList.of(m("a", m("b", m("c1", 3, "d", "zz")))));
+  }
+
+  @Test
+  @Disabled // TODO: array globs handling needs to be refactored in DocumentService
```

Comment:
Unfortunately array globs do not work yet due to the intricate interplay of new and old field handling code (in `DocumentService`). I suppose we'll have to fix this in a follow-up PR as it looks like a large refactoring.

Reply:
Yea this sounds good - iirc Ivan also fixed bugs in his PR that added array globbing, so it will be good to have that follow-up :)

Reply:
Actually, I fixed this in 5b38675a223c3e3c9ade9342b987422b8ee277b8